client/src/components/Profile.jsx

```diff
@@ -22,13 +22,23 @@ function Profile() {
   const { getProfile } = useContext(AuthContext);
   const handleOpen = () => setOpen(true);
   const handleClose = () => setOpen(false);
+  const location = useLocation();
+  const { id } = location.state;
 
   useEffect(() => {
-    getProfile('wTNTNk9WndF91iuHDyym')
-      .then((userProfile) => {
-        setProfile(userProfile.data());
-      })
-      .catch((err) => err);
+    if (id) {
```

Comment:
Could this if else statement be simplified? It looks like the only thing that's changing is the id. Maybe just a || fallback could work